build.rs

```diff
@@ -1,39 +1,46 @@
-extern crate gcc;
+extern crate cc;
 
 use std::env;
 
 fn main() {
     let target = env::var("TARGET").unwrap();
-    let mut base_config = gcc::Build::new();
-    base_config.cpp(true)
+    let mut base_config = cc::Build::new();
+    base_config
+        .cpp(true)
         .include("bitcoin/src")
         .include("bitcoin/src/secp256k1/include")
-        .flag("-g").flag("-O2").flag("-std=c++11").flag("-Wno-unused-parameter")
+        .opt_level(2)
+        .debug(true)
+        .flag_if_supported("-std=c++11")
```

Comment:
Is there some quivalent for windows? I dont think we want to use whatever default version of C++ the compiler supports.